### ISSUE TITLE
Export

### ISSUE THREAD

**Issue by Sven-Olav** [gruijter/com.gruijter.insights2csv#19]
I have 2 issues:
1. If i export last month I will not get export from all apps, for example fibaro. The only one that work is if I choose last 2 years. Worked fine in the beginning but approx a year ago this happened.
2. When I export, sometimes there is values in between that obviously is incorrect.

Any suggestion on how to fix this will be appreciated:)

**Comment by gruijter** [comment]
> 1. f i export last month I will not get export from all apps, for example fibaro. The only one that work is if I choose last 2 years. Worked fine in the beginning but approx a year ago this happened.

This can happen if the Insights database on Homey is corrupt. What happens if you try to create insights graphs for Fibaro for all available variables. (via https://insights.homey.app/) ? Does that work correctly?

If it is corrupt maybe making a backup of your Homey, and then restoring it again might fix it. Otherwise you could simply delete all the Insights from the Fibaro app (also via insights - clear data)




> 2\. When I export, sometimes there is values in between that obviously is incorrect.

Do you have an example? Note that Homey is not storing the actual values in the Insights database. It stores **averages** of the data to compress the storage capacity. So if for instance your device reports 1watt and later 2watt, Homey would store an average number of 1.5 watts.

**Comment by Sven-Olav** [comment]
Hi again, and thanks for quick answer😀

1. it seems to work fine:


The first picture is a fibaro unit that i dont Get any data from, exept if I use last 2 year…

The second picture is the error with wrong data, it seems like it is homey providing that data - you see one value that is way above the rest…

Mvh

Sven-Olav

**Comment by gruijter** [comment]
Mmh... If you do get Insights.homey.app  displayed correctly I can only assume that the API I am using to fetch the data has a bug. Athom recently deprecated the existing API and released a new API for developers. But before I can implement this new API I will have to do significant rewriting of the app and migrate it to SDK3. I have it on my roadmap to do that, but it will take quite some (spare)time...

So sit tight and hold on for a while :)

**Comment by gruijter** [comment]
Please try the new version 3:
https://homey.app/en-gb/app/com.gruijter.insights2csv/Archive-Insights/test/

Reopen the issue if you still have this problem

**Comment by Sven-Olav** [comment]
Hi and thanks for the Feedback. I tried to install the new version, but I only recieve data from yesterday 2200. This applies to all my apps…
Hope You can fix it - or should i roll back to 2.9.2?
I also Get an error in the log saying something about corrupt data, but that doesn’t apply to all apps, see screenshot.


Mvh

Sven-Olav

**Comment by gruijter** [comment]
Can you send me an app diagnostics report?

**Comment by Sven-Olav** [comment]
8b9d3657-34ba-46d3-8722-6f0ceae1ead2

Yes, and This is the code😀

Mvh

Sven-Olav

**Comment by gruijter** [comment]
Actually in the log all looks good, except there was an occasional error, that I think is related to accessing your NAS via SMB:
```Error: connect EALREADY 10.0.0.180:445 - Local (10.0.0.134:48080)
    at internalConnect (net.js:917:16)
    at defaultTriggerAsyncIdScope (internal/async_hooks.js:311:12)
    at net.js:1007:9
    at processTicksAndRejections (internal/process/task_queues.js:79:11) {
  errno: 'EALREADY',
  code: 'EALREADY',
  syscall: 'connect',
  address: '10.0.0.180',
  port: 445
}
```

Not sure why that happens, but all exports seem good, and no other errors occur.

**Comment by Sven-Olav** [comment]
That is the IP address of my nas, so that is correct.
I will try again later and if it still doesn’t work I will roll back to 2.9.2

Mvh

Sven-Olav

**Comment by Sven-Olav** [comment]
So I tried again, and today I got data from 22:00 yesterday and until now, despite that i choose «this year».
Yesterday i got data from 22:00 the day before yesterday….

I hope you have sn suggestion for me:)

nr

Sven-Olav

**Comment by gruijter** [comment]
can confirm this is a bug in homey-api. I contacted Athom and am waiting their response / resolvement.

**Comment by gruijter** [comment]
fixed in v3.0.1 https://homey.app/a/com.gruijter.insights2csv/test/

**Comment by Sven-Olav** [comment]
Thanks for the update, I’m looking forward to hear from you again😀

Mvh

Sven-Olav
